src/Symfony/Component/DependencyInjection/Loader/Configurator/ContainerConfigurator.php

```diff
@@ -36,25 +41,63 @@ class ContainerConfigurator extends AbstractConfigurator
     private $file;
     private $anonymousCount = 0;
     private $env;
+    private $generator;
 
-    public function __construct(ContainerBuilder $container, PhpFileLoader $loader, array &$instanceof, string $path, string $file, string $env = null)
+    public function __construct(ContainerBuilder $container, PhpFileLoader $loader, array &$instanceof, string $path, string $file, string $env = null, GeneratorInterface $generator = null)
     {
         $this->container = $container;
         $this->loader = $loader;
         $this->instanceof = &$instanceof;
         $this->path = $path;
         $this->file = $file;
         $this->env = $env;
+        $this->generator = $generator ?? new ConfigBuilderGenerator();
     }
 
     final public function extension(string $namespace, array $config)
+    {
+        $this->throwIfExtensionDoesNotExist($namespace);
+
+        $this->container->loadFromExtension($namespace, static::processValue($config));
+    }
+
+    /**
+     * @param string $namespace example extension alias ("framework") or FQCN string
+     *                          for a class implementing ConfigBuilderInterface
+     */
+    final public function configBuilder(string $namespace): ConfigBuilderInterface
+    {
+        if (!class_exists(ConfigBuilderGenerator::class)) {
```

Comment:
If class didn't exists,  you would have an exception in the constructor ;-)

Reply:
I dont think so. 
You have have symfony/dependency-injection:5.3 and symfony/config: 5.2 and not use the config builder. That is why I dont want the exception in the constructor.

Reply:
With config 5.2 and di 5.3, a user may do: 

```
use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;

return static function (ContainerConfigurator $container) {

    // This should throw a nice exception
    $security = $container->configBuilder('security');

   // This is valid
   $container->extension('security', [/* .. */]);
}
```

Throwing in constructor would disable the valid use case. Not throwing at all would lead to weird PHP exceptions.

Reply:
Oh no.. Im sorry. I misunderstood you both. 

Give me a moment to fix this.

Reply:
Yes @jderusse you are correct. I misread your comment and though you suggested to add the same check in the constructor instead. 

I now only create a `new ConfigBuilderGenerator` after I check if the class exist. 

Thank you for being patient with me.